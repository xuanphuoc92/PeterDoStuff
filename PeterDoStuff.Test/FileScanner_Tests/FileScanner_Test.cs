using ApprovalTests.Reporters;
using FluentAssertions;
using PeterDoStuff.Test.Extensions;
using PeterDoStuff.Tools;

namespace PeterDoStuff.Test.FileScanner_Tests
{
    [TestClass]
    [UseReporter(typeof(DiffReporter))]
    public class FileScanner_Test
    {
        [TestMethod]
        public void _01_ScanZipFile()
        {
            using var scanner = new FileScanner();
            var result = scanner.ScanZip("FileScanner_Tests\\TestZipFile.zip");
            result.Should().Be(FileScanner.SUCCESSFUL);
            scanner.ZipStats.Count.Should().Be(4);
            string csvContent = scanner.ToCsv();
            csvContent.Verify();
        }

        [TestMethod]
        public void _02_ScanNonExistFile()
        {
            using var scanner = new FileScanner();
            var result = scanner.ScanZip("NonExistFile.zip");
            result.Should().NotBe(FileScanner.SUCCESSFUL);
            result.WriteToConsole();
        }

        [TestMethod]
        public void _03_ScanNonZipFile()
        {
            using var scanner = new FileScanner();
            var result = scanner.ScanZip("FileScanner_Tests\\TestTextFile.txt");
            result.Should().NotBe(FileScanner.SUCCESSFUL);
            result.WriteToConsole();
        }
    }
}
