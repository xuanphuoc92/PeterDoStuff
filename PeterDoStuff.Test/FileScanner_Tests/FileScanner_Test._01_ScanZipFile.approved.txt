File Path,Extension,Number Of Lines
SubFolder/Test,Test,2
SubFolder/TestTextFile.csv,.csv,3
Test,Test,2
TestTextFile.txt,.txt,3
